cunumeric/module.py

```diff
@@ -1285,26 +1287,71 @@ def check_shape_dtype(
     return converted, ArrayInfo(ndim, shape, dtype)
 
 
-def _block(arr, cur_depth, depth):
+def _block_collect_slices(arr, cur_depth, depth):
+    # collects slices for each array in `arr`
+    # the last outcome will be slices on every dimension of the output array
+    # for each array in `arr`
     if cur_depth < depth:
-        inputs = list(_block(each, cur_depth + 1, depth) for each in arr)
+        sublist_results = list(
+            _block_collect_slices(each, cur_depth + 1, depth) for each in arr
+        )
+        arrays, outshape_list, slices = zip(*sublist_results)
```

Comment:
is this really correct? `sublist_results` is a list of 3-tuples, and this zipping will return another list of 3 n-tuples where n is `len(sublist_results)`. Isn't "unzipping `sublist_results`" what you want? I think I don't understand how this code worked.